geode-core/src/main/java/org/apache/geode/pdx/internal/PdxInstanceImpl.java

```diff
@@ -464,8 +465,14 @@ public String toString() {
       result.append(fieldType.getFieldName());
       result.append("=");
       try {
-        // TODO check to see if getField returned an array and if it did use Arrays.deepToString
-        result.append(ur.readField(fieldType.getFieldName()));
+        final Object value = ur.readField(fieldType.getFieldName());
+        if (value instanceof byte[]) {
+          result.append(Hex.toHex((byte[]) value));
+        } else if (value.getClass().isArray()) {
+          result.append(Arrays.deepToString((Object[]) value));
```

Comment:
I think this cast might break if the array is not an Object[] array - what if it is an int[] or a Person[], or ???